src/contracts/pods/EigenPodManager.sol

```diff
@@ -134,68 +199,131 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
 
-    /**
-     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
-     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
-     */
-    function decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
-        external onlyStrategyManager onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+        _removeShares(slashedPodOwner, shareAmount);
+        _withdrawRestakedBeaconChainETH(slashedPodOwner, slashedFundsRecipient, shareAmount);
+    }
+
+    function slashQueuedWithdrawal(
+        address slashedFundsRecipient,
+        BeaconChainQueuedWithdrawal memory queuedWithdrawal
+    )
+        external 
+        onlyOwner 
+        onlyFrozen(queuedWithdrawal.delegatedAddress) 
+        nonReentrant 
     {
-        ownerToPod[podOwner].decrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        // find the withdrawalRoot
+        bytes32 withdrawalRoot = calculateWithdrawalRoot(queuedWithdrawal);
+
+         // verify that the queued withdrawal is pending
+        require(
+            withdrawalRootPending[withdrawalRoot],
+            "EigenPodManager.slashQueuedWithdrawal: withdrawal is not pending"
+        );
+
+        // reset the storage slot in mapping of queued withdrawals
+        withdrawalRootPending[withdrawalRoot] = false;
+
+        _withdrawRestakedBeaconChainETH(queuedWithdrawal.podOwner, slashedFundsRecipient, queuedWithdrawal.shares);
     }
 
     /**
-     * @notice This function is called to increment withdrawableRestakedExecutionLayerGwei when a validator's withdrawal is completed.
-     * @param amountWei is the amount of ETH in wei to increment withdrawableRestakedExecutionLayerGwei by
+     * @notice forces the podOwner into the "undelegation limbo" mode
+     * @param podOwner is the staker to be forced into undelegation limbo
+     * @dev This function can only be called by the DelegationManager contract
      */
-    function incrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
-        external onlyStrategyManager onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+    function forceIntoUndelegationLimbo(address podOwner) external 
+        onlyDelegationManager 
+        onlyWhenNotPaused(PAUSED_WITHDRAWALS)
+        onlyNotFrozen(podOwner)
+        nonReentrant
     {
-        ownerToPod[podOwner].incrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        if (!stakerHasNoDelegatedShares(podOwner)) {
+            _enterUndelegationLimbo(podOwner);
+        }
     }
 
+    /**
+     * Sets the maximum number of pods that can be deployed
+     * @param newMaxPods The new maximum number of pods that can be deployed
+     * @dev Callable by the unpauser of this contract
+     */
+    function setMaxPods(uint256 newMaxPods) external onlyUnpauser {
```

Comment:
can we put this near the top of the external functions?

Reply:
I think this grouping makes more sense, with all the can-only-be-called-by-a-single-address functions grouped at the bottom of the external functions https://github.com/Layr-Labs/eigenlayer-contracts/pull/135/commits/4c6168d5cbc1d95c7accc14b283b88606f9d2723